core/src/main/java/com/linecorp/armeria/client/endpoint/SlowStartAwareEndpointSelectionStrategy.java

```diff
@@ -0,0 +1,135 @@
+                                                 BiFunction<Endpoint, Duration, Integer> endpointWeighter) {
+            this.endpointGroup = endpointGroup;
+            this.endpointWeighter = endpointWeighter;
+            final List<Endpoint> endpoints = endpointGroup.endpoints();
+            endpointsAndWeights = new EndpointsAndWeights(endpoints);
+            previousEndoints = endpoints;
+            endpointGroup.addListener(newEndpoints -> {
+                endpointsAndWeights = new EndpointsAndWeights(newEndpoints);
+                scheduleUpdateRequestWeight(newEndpoints, previousEndoints);
+                previousEndoints = endpoints;
+            });
+        }
+
+        @Override
+        public EndpointGroup group() {
+            return endpointGroup;
+        }
+
+        @Override
+        public EndpointSelectionStrategy strategy() {
+            return SlowStartAwareEndpointSelectionStrategy.this;
+        }
+
+        @Override
+        public Endpoint select(ClientRequestContext ctx) {
+            final int currentSequence = sequence.getAndIncrement();
+            return endpointsAndWeights.selectEndpoint(currentSequence);
+        }
+
+        private void scheduleUpdateRequestWeight(List<Endpoint> currentlyEndpoints,
+                                                 List<Endpoint> previousEndpoints) {
+            if (slowStartInterval.isZero()) {
+                return;
+            }
+            if (updateEndpointWeightFuture != null && !updateEndpointWeightFuture.isDone()) {
+                updateEndpointWeightFuture.cancel(true);
+            }
+            final Set<Endpoint> newEndpoints = Sets.difference(ImmutableSet.copyOf(previousEndpoints),
+                                                               ImmutableSet.copyOf(currentlyEndpoints));
+            final int step = Math.max((int) slowStartInterval.toMillis() / maxWeight, 1);
+
+            updateEndpointWeightFuture = clientFactory.eventLoopGroup()
+                                                      .schedule(() -> updateEndpointWeight(currentlyEndpoints,
+                                                                                           newEndpoints, step),
+                                                                slowStartInterval.toMillis() / maxWeight,
+                                                                TimeUnit.MILLISECONDS);
+        }
+
+        private void updateEndpointWeight(List<Endpoint> endpoints, Set<Endpoint> newEndpoints, int step) {
+            endpointsAndWeights = new EndpointsAndWeights(
+                    endpoints.stream()
+                             .map(e -> {
+                                 if (e.weight() >= maxWeight || !newEndpoints.contains(e)) {
+                                     return e;
+                                 }
+                                 return e.withWeight(Math.min(e.weight() + step, maxWeight));
```

Comment:
The weight of each endpoint has to increase proportionally. For example, if we increase the weight for 10 seconds, two endpoints with the weight of 500 and 1000 should have the weight of 250 and 500 at 5-second point.